bears/vcs/VCSCommitBear.py

```diff
@@ -0,0 +1,140 @@
+import abc
+import re
+
+from coalib.bears.GlobalBear import GlobalBear
+from coalib.results.HiddenResult import HiddenResult
+from coalib.misc.Shell import run_shell_command
+from coala_utils.decorators import (enforce_signature, generate_ordering,
+                                    generate_repr)
+
+
+@generate_repr(('id', hex),
+               'origin',
+               'raw_commit_message',
+               'commit_sha',
+               'commit_type',
+               'modified_files',
+               'added_files',
+               'deleted_files',
+               'message')
+@generate_ordering('raw_commit_message',
+                   'commit_sha',
+                   'commit_type',
+                   'modified_files',
+                   'added_files',
+                   'deleted_files',
+                   'severity',
+                   'confidence',
+                   'origin',
+                   'message_base',
+                   'message_arguments',
+                   'aspect',
+                   'additional_info',
+                   'debug_msg')
+class CommitResult(HiddenResult):
+
+    @enforce_signature
+    def __init__(self, origin,
+                 raw_commit_message: str,
+                 commit_sha: str,
+                 commit_type: list,
+                 modified_files: list,
+                 added_files: list,
+                 deleted_files: list):
+
+        HiddenResult.__init__(self, origin, '')
+        self.raw_commit_message = raw_commit_message
+        self.commit_sha = commit_sha
+        self.commit_type = commit_type
+        self.modified_files = modified_files
+        self.added_files = added_files
+        self.deleted_files = deleted_files
+        self.message = 'HEAD commit information'
+
+
+class VCSCommitBear(GlobalBear):
+    __metaclass__ = abc.ABCMeta
+    LANGUAGES = {'Git'}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+
+    @abc.abstractmethod
+    def get_head_commit(self):
+        """
+        Return the commit message from the head commit
+        """
+
+    def analyze_git_commit(self, head_commit):
+        """
+        Check the current commit at HEAD.
+
+        Return the commit information such as commit message,
+        type of commit, commit SHA, modified, added and
+        delete files by the commit.
```

Comment:
Also explain about the paremeters and the return value, see other bears, c.f `CommitBear.py`